x-pack/plugins/search_playground/server/lib/elasticsearch_retriever.ts

```diff
@@ -82,13 +83,21 @@ export class ElasticsearchRetriever extends BaseRetriever {
 
       // default elasticsearch doc to LangChain doc
       let mapper: HitDocMapper = (hit: SearchHit<any>) => {
-        const pageContentFieldKey =
+        let pageContentFieldKey =
           typeof this.content_field === 'string'
             ? this.content_field
             : this.content_field[hit._index as string];
 
+        // field values can be a single item array or plain text, that is why needed to check the types before accessing the field
+        pageContentFieldKey = Array.isArray(pageContentFieldKey)
```

Comment:
@joemcelroy: I am getting two sets of input into this function.
- Getting object notation `{ index: 'field', website: 'body_content' }` from [conversation_chain.test.ts](https://github.com/elastic/kibana/blob/afaba520d08e470f3350a8afeb02cd9bc574d6dc/x-pack/plugins/search_playground/server/lib/conversational_chain.test.ts)
- From UI Array notation `{'index_name' :['field_name`]}`

Please let me know if this is expected.

Reply:
the retriever should expect a type of `Record<string, string>` - an object where the key is the index name and the value is the field name.

UI is built for a time when we have multiple fields that can be selected. What should happen is that the UI input is converted into the above, picking the first field in the array. This should be updated in the route.ts.

Im surprised the route didn't throw an API validation error. Its currently only accepting a string.